float32 x
float32 y
float32 z
float32 rx
float32 ry
float32 rz